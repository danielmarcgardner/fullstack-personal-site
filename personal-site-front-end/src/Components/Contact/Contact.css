#contacthero {
  width: 100%;
  height: fit-content;
  background-color: #f2cfa2;
  color: #4c4a47;
}

#contactheroheader {
  font-family: 'Just Another Hand', cursive;
  font-size: 48px;
}

.contactdivs {
  text-align: center;
  font-family: 'Just Another Hand', cursive;
  border: 2px solid #4c4a47;
  border-radius: 10%;
  width: 200px;
}

.contactdivs h3 {
  font-family: 'Just Another Hand', cursive;
  font-size: 36px;
  color: #4c4a47;
}

.contactdivs p {
  font-size: 28px;
  color: #4c4a47;
}

.contactdivs:hover {
  background-color: #4c4a47;

}
.contactdivs:hover a h3, .contactdivs:hover a p {
  color: #f2cfa2 !important;
}
