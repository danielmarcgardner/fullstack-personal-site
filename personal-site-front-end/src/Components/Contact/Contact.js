import React from 'react';
import { Container, Grid, Icon } from 'semantic-ui-react';
import './Contact.css';

const Contact = () => (
  <div id="contacthero">
    <Container>
      <h1 id="contactheroheader">Contact</h1>
      <Grid columns={2} centered textAlign="center" stackable>
        <Grid.Row>
          <Grid.Column computer={4} tablet={8} mobile={16}>
            <div className="contactdivs">
              <a href="mailto:daniel.marc.gardner@gmail.com">
                <h3>Email Me</h3>
                <Icon name="mail" size="huge" className="contacticon" />
                <p>daniel.marc.gardner@gmail.com</p>
              </a>
            </div>
          </Grid.Column>
          <Grid.Column computer={4} tablet={8} mobile={16}>
            <div className="contactdivs">
              <a href="https://github.com/danielmarcgardner">
                <h3>GitHub</h3>
                <Icon name="github" size="huge" className="contacticon" />
                <p>@danielmarcgardner</p>
              </a>
            </div>
          </Grid.Column>
          <Grid.Column computer={4} tablet={8} mobile={16}>
            <div className="contactdivs">
              <a href="https://www.linkedin.com/in/danielmarcgardner/">
                <h3>LinkedIn</h3>
                <Icon name="linkedin" size="huge" className="contacticon" />
                <p>@danielmarcgardner</p>
              </a>
            </div>
          </Grid.Column>
          <Grid.Column computer={4} tablet={8} mobile={16}>
            <div className="contactdivs">
              <a href="https://twitter.com/danielmgardner1">
                <h3>Twitter</h3>
                <Icon name="twitter" size="huge" className="contacticon" />
                <p>@danielmgardner1</p>
              </a>
            </div>
          </Grid.Column>
        </Grid.Row>
        <p id="copyright">&copy; Daniel Gardner 2017</p>
      </Grid>
    </Container>
  </div>
);

export default Contact;
