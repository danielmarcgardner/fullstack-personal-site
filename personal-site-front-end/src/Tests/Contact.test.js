import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Contact from '../Components/Contact/Contact';

const component = shallow(<Contact />);

describe('Contact Hero Div', () => {
  it('Renders', () => {
    expect(component).toMatchSnapshot();
  });
  it('Has 4 contactdivs', () => {
    expect(component.find('.contactdivs').length).toEqual(4);
  });
  it('Has links to the correct places', () => {
    expect(component.find('a').nodes[0].props.href).toEqual('mailto:daniel.marc.gardner@gmail.com');
    expect(component.find('a').nodes[1].props.href).toEqual('https://github.com/danielmarcgardner');
    expect(component.find('a').nodes[2].props.href).toEqual('https://www.linkedin.com/in/danielmarcgardner/');
    expect(component.find('a').nodes[3].props.href).toEqual('https://twitter.com/danielmgardner1');
  });
});
