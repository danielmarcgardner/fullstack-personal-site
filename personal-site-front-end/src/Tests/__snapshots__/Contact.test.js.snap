// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Hero Div Renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    id="contacthero"
>
    <Container>
        <h1
            id="contactheroheader"
        >
            Contact
        </h1>
        <Grid
            centered={true}
            columns={2}
            stackable={true}
            textAlign="center"
        >
            <GridRow>
                <GridColumn
                    computer={4}
                    mobile={16}
                    tablet={8}
                >
                    <div
                        className="contactdivs"
                    >
                        <a
                            href="mailto:daniel.marc.gardner@gmail.com"
                        >
                            <h3>
                                Email Me
                            </h3>
                            <Icon
                                as="i"
                                className="contacticon"
                                name="mail"
                                size="huge"
                            />
                            <p>
                                daniel.marc.gardner@gmail.com
                            </p>
                        </a>
                    </div>
                </GridColumn>
                <GridColumn
                    computer={4}
                    mobile={16}
                    tablet={8}
                >
                    <div
                        className="contactdivs"
                    >
                        <a
                            href="https://github.com/danielmarcgardner"
                        >
                            <h3>
                                GitHub
                            </h3>
                            <Icon
                                as="i"
                                className="contacticon"
                                name="github"
                                size="huge"
                            />
                            <p>
                                @danielmarcgardner
                            </p>
                        </a>
                    </div>
                </GridColumn>
                <GridColumn
                    computer={4}
                    mobile={16}
                    tablet={8}
                >
                    <div
                        className="contactdivs"
                    >
                        <a
                            href="https://www.linkedin.com/in/danielmarcgardner/"
                        >
                            <h3>
                                LinkedIn
                            </h3>
                            <Icon
                                as="i"
                                className="contacticon"
                                name="linkedin"
                                size="huge"
                            />
                            <p>
                                @danielmarcgardner
                            </p>
                        </a>
                    </div>
                </GridColumn>
                <GridColumn
                    computer={4}
                    mobile={16}
                    tablet={8}
                >
                    <div
                        className="contactdivs"
                    >
                        <a
                            href="https://twitter.com/danielmgardner1"
                        >
                            <h3>
                                Twitter
                            </h3>
                            <Icon
                                as="i"
                                className="contacticon"
                                name="twitter"
                                size="huge"
                            />
                            <p>
                                @danielmgardner1
                            </p>
                        </a>
                    </div>
                </GridColumn>
            </GridRow>
        </Grid>
    </Container>
</div>,
  "nodes": Array [
    <div
      id="contacthero"
>
      <Container>
            <h1
                  id="contactheroheader"
            >
                  Contact
            </h1>
            <Grid
                  centered={true}
                  columns={2}
                  stackable={true}
                  textAlign="center"
            >
                  <GridRow>
                        <GridColumn
                              computer={4}
                              mobile={16}
                              tablet={8}
                        >
                              <div
                                    className="contactdivs"
                              >
                                    <a
                                          href="mailto:daniel.marc.gardner@gmail.com"
                                    >
                                          <h3>
                                                Email Me
                                          </h3>
                                          <Icon
                                                as="i"
                                                className="contacticon"
                                                name="mail"
                                                size="huge"
                                          />
                                          <p>
                                                daniel.marc.gardner@gmail.com
                                          </p>
                                    </a>
                              </div>
                        </GridColumn>
                        <GridColumn
                              computer={4}
                              mobile={16}
                              tablet={8}
                        >
                              <div
                                    className="contactdivs"
                              >
                                    <a
                                          href="https://github.com/danielmarcgardner"
                                    >
                                          <h3>
                                                GitHub
                                          </h3>
                                          <Icon
                                                as="i"
                                                className="contacticon"
                                                name="github"
                                                size="huge"
                                          />
                                          <p>
                                                @danielmarcgardner
                                          </p>
                                    </a>
                              </div>
                        </GridColumn>
                        <GridColumn
                              computer={4}
                              mobile={16}
                              tablet={8}
                        >
                              <div
                                    className="contactdivs"
                              >
                                    <a
                                          href="https://www.linkedin.com/in/danielmarcgardner/"
                                    >
                                          <h3>
                                                LinkedIn
                                          </h3>
                                          <Icon
                                                as="i"
                                                className="contacticon"
                                                name="linkedin"
                                                size="huge"
                                          />
                                          <p>
                                                @danielmarcgardner
                                          </p>
                                    </a>
                              </div>
                        </GridColumn>
                        <GridColumn
                              computer={4}
                              mobile={16}
                              tablet={8}
                        >
                              <div
                                    className="contactdivs"
                              >
                                    <a
                                          href="https://twitter.com/danielmgardner1"
                                    >
                                          <h3>
                                                Twitter
                                          </h3>
                                          <Icon
                                                as="i"
                                                className="contacticon"
                                                name="twitter"
                                                size="huge"
                                          />
                                          <p>
                                                @danielmgardner1
                                          </p>
                                    </a>
                              </div>
                        </GridColumn>
                  </GridRow>
            </Grid>
      </Container>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Contact />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          id="contacthero"
>
          <Container>
                    <h1
                              id="contactheroheader"
                    >
                              Contact
                    </h1>
                    <Grid
                              centered={true}
                              columns={2}
                              stackable={true}
                              textAlign="center"
                    >
                              <GridRow>
                                        <GridColumn
                                                  computer={4}
                                                  mobile={16}
                                                  tablet={8}
                                        >
                                                  <div
                                                            className="contactdivs"
                                                  >
                                                            <a
                                                                      href="mailto:daniel.marc.gardner@gmail.com"
                                                            >
                                                                      <h3>
                                                                                Email Me
                                                                      </h3>
                                                                      <Icon
                                                                                as="i"
                                                                                className="contacticon"
                                                                                name="mail"
                                                                                size="huge"
                                                                      />
                                                                      <p>
                                                                                daniel.marc.gardner@gmail.com
                                                                      </p>
                                                            </a>
                                                  </div>
                                        </GridColumn>
                                        <GridColumn
                                                  computer={4}
                                                  mobile={16}
                                                  tablet={8}
                                        >
                                                  <div
                                                            className="contactdivs"
                                                  >
                                                            <a
                                                                      href="https://github.com/danielmarcgardner"
                                                            >
                                                                      <h3>
                                                                                GitHub
                                                                      </h3>
                                                                      <Icon
                                                                                as="i"
                                                                                className="contacticon"
                                                                                name="github"
                                                                                size="huge"
                                                                      />
                                                                      <p>
                                                                                @danielmarcgardner
                                                                      </p>
                                                            </a>
                                                  </div>
                                        </GridColumn>
                                        <GridColumn
                                                  computer={4}
                                                  mobile={16}
                                                  tablet={8}
                                        >
                                                  <div
                                                            className="contactdivs"
                                                  >
                                                            <a
                                                                      href="https://www.linkedin.com/in/danielmarcgardner/"
                                                            >
                                                                      <h3>
                                                                                LinkedIn
                                                                      </h3>
                                                                      <Icon
                                                                                as="i"
                                                                                className="contacticon"
                                                                                name="linkedin"
                                                                                size="huge"
                                                                      />
                                                                      <p>
                                                                                @danielmarcgardner
                                                                      </p>
                                                            </a>
                                                  </div>
                                        </GridColumn>
                                        <GridColumn
                                                  computer={4}
                                                  mobile={16}
                                                  tablet={8}
                                        >
                                                  <div
                                                            className="contactdivs"
                                                  >
                                                            <a
                                                                      href="https://twitter.com/danielmgardner1"
                                                            >
                                                                      <h3>
                                                                                Twitter
                                                                      </h3>
                                                                      <Icon
                                                                                as="i"
                                                                                className="contacticon"
                                                                                name="twitter"
                                                                                size="huge"
                                                                      />
                                                                      <p>
                                                                                @danielmgardner1
                                                                      </p>
                                                            </a>
                                                  </div>
                                        </GridColumn>
                              </GridRow>
                    </Grid>
          </Container>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          id="contacthero"
>
          <Container>
                    <h1
                              id="contactheroheader"
                    >
                              Contact
                    </h1>
                    <Grid
                              centered={true}
                              columns={2}
                              stackable={true}
                              textAlign="center"
                    >
                              <GridRow>
                                        <GridColumn
                                                  computer={4}
                                                  mobile={16}
                                                  tablet={8}
                                        >
                                                  <div
                                                            className="contactdivs"
                                                  >
                                                            <a
                                                                      href="mailto:daniel.marc.gardner@gmail.com"
                                                            >
                                                                      <h3>
                                                                                Email Me
                                                                      </h3>
                                                                      <Icon
                                                                                as="i"
                                                                                className="contacticon"
                                                                                name="mail"
                                                                                size="huge"
                                                                      />
                                                                      <p>
                                                                                daniel.marc.gardner@gmail.com
                                                                      </p>
                                                            </a>
                                                  </div>
                                        </GridColumn>
                                        <GridColumn
                                                  computer={4}
                                                  mobile={16}
                                                  tablet={8}
                                        >
                                                  <div
                                                            className="contactdivs"
                                                  >
                                                            <a
                                                                      href="https://github.com/danielmarcgardner"
                                                            >
                                                                      <h3>
                                                                                GitHub
                                                                      </h3>
                                                                      <Icon
                                                                                as="i"
                                                                                className="contacticon"
                                                                                name="github"
                                                                                size="huge"
                                                                      />
                                                                      <p>
                                                                                @danielmarcgardner
                                                                      </p>
                                                            </a>
                                                  </div>
                                        </GridColumn>
                                        <GridColumn
                                                  computer={4}
                                                  mobile={16}
                                                  tablet={8}
                                        >
                                                  <div
                                                            className="contactdivs"
                                                  >
                                                            <a
                                                                      href="https://www.linkedin.com/in/danielmarcgardner/"
                                                            >
                                                                      <h3>
                                                                                LinkedIn
                                                                      </h3>
                                                                      <Icon
                                                                                as="i"
                                                                                className="contacticon"
                                                                                name="linkedin"
                                                                                size="huge"
                                                                      />
                                                                      <p>
                                                                                @danielmarcgardner
                                                                      </p>
                                                            </a>
                                                  </div>
                                        </GridColumn>
                                        <GridColumn
                                                  computer={4}
                                                  mobile={16}
                                                  tablet={8}
                                        >
                                                  <div
                                                            className="contactdivs"
                                                  >
                                                            <a
                                                                      href="https://twitter.com/danielmgardner1"
                                                            >
                                                                      <h3>
                                                                                Twitter
                                                                      </h3>
                                                                      <Icon
                                                                                as="i"
                                                                                className="contacticon"
                                                                                name="twitter"
                                                                                size="huge"
                                                                      />
                                                                      <p>
                                                                                @danielmgardner1
                                                                      </p>
                                                            </a>
                                                  </div>
                                        </GridColumn>
                              </GridRow>
                    </Grid>
          </Container>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Contact />,
}
`;
